pkg/kubelet/kubelet.go

```diff
@@ -81,6 +84,7 @@ type SyncHandler interface {
 	// Syncs current state to match the specified pods. SyncPodType specified what
 	// type of sync is occuring per pod. StartTime specifies the time at which
 	// syncing began (for use in monitoring).
+	SyncDockerPods(pods []api.BoundPod, podSyncTypes map[types.UID]metrics.SyncPodType, startTime time.Time) error
```

Comment:
Why don't you just have a docker and non-docker implementation of this `SyncHandler` interface? Then inject the relevant one to `kl.syncLoop`.